### PR TITLE
[CDAP-20470] - Unit tests for streaming event functions.

### PR DESCRIPTION
**Pull request by greeshmaswaminathan**
[CDAP-20470](https://cdap.atlassian.net/browse/CDAP-20470) : Unit tests for changes in https://github.com/cdapio/cdap/pull/15074 .

[CDAP-20470]: https://cdap.atlassian.net/browse/CDAP-20470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/spark/streaming/MockStreamingEventSource.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+  /**
+   * Config for mock source.
+   */
+  private static class Conf extends PluginConfig {
+
+    private String schema;
+    private String records;
+    @Nullable
+    private Long intervalMillis;
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/spark/streaming/MockStreamingEventSource.java]: this looks unused, unless I'm missing something?

File: cdap-app-templates/cdap-etl/cdap-data-streams-base/src/test/java/io/cdap/cdap/datastreams/DataStreamsTest.java [hunk 2]
```diff
@@ -1611,4 +1614,190 @@ public void testRetryTimeout() throws Exception {
...
+      Tasks.waitFor(true,
+          () -> {
+            Optional<byte[]> startedState = state.getState(startedKey);
+            Optional<byte[]> completedState = state.getState(completedKey);
+            Optional<byte[]> retryState = state.getState(retryKey);
+            return startedState.isPresent() && completedState.isPresent()
+                && !retryState.isPresent();
+          },
+          3, TimeUnit.MINUTES);
+      state.deleteSate(completedKey);
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-data-streams-base/src/test/java/io/cdap/cdap/datastreams/DataStreamsTest.java:1677]: typo: deleteSate -> deleteState

can fix in a separate PR

Comment (greeshmaswaminathan) [reply cdap-app-templates/cdap-etl/cdap-data-streams-base/src/test/java/io/cdap/cdap/datastreams/DataStreamsTest.java:1677]: Ack, will create a new PR for this.

File: cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/spark/streaming/MockStreamingEventSource.java (hunk 1 above)
Comment (greeshmaswaminathan) [reply cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/spark/streaming/MockStreamingEventSource.java]: Removed all unused config properties.

**Review by albertshau — APPROVED** [review]

**Pull request merged**